include/swift/Runtime/Metadata.h

```diff
@@ -878,7 +878,11 @@ const TypeContextDescriptor *swift_getTypeContextDescriptor(const Metadata *type
 SWIFT_RUNTIME_EXPORT
 const HeapObject *swift_getKeyPath(const void *pattern, const void *arguments);
 
-#if defined(swiftCore_EXPORTS)
+// For whatever reason, MSVC is fine with these declarations unless
+// they're dllimport, at which point it complains that they don't have
+// a C-compatible type.
```

Comment:
These definitions cannot be `dllimport` by definition.  `dllimport` means that they are homed elsewhere and it is a declaration only.  I think that this needs to be fixed.  Definitions in headers shouldn't be exported nor imported, they are universally emitted.

Reply:
These aren't definitions, they're just declarations.  I tried to change the `#if` condition so that these would properly be declared and usable in all libraries, but MSVC complained about an `extern "C"` function with a non-C++ type.  I don't know why MSVC only complains about that in other libraries, but I assume it has something to do with `dllimport`.  Anyway, the actual impact of this change (besides renaming the `#define`) is just that I added a comment to try to explain why these are only conditionally declared.

Reply:
I would love to know why this `#if` is necessary and find a way to fix it.